docs/pallas/tpu/pipelining.md

```diff
@@ -0,0 +1,397 @@
+
+Let's talk about the components of this diagram in more detail:
+
+* **Memory spaces**: A TPU has high-bandwidth memory (HBM) which is what we often think of as "device memory". There is also vector memory (VMEM), a cache meant for storing vector and array values, and scalar memory (SMEM), a cache designed to store scalar values.
+* **Registers**: A TensorCore has two main types of registers: vector registers (VREGs) store array values, and scalar registers (SREGs) store scalar values. Values can be loaded into memory from their respective caches (VMEM for VREGs and SMEM for SREGs).
+* **Compute units**: A TensorCore has a scalar unit, vector unit (VPU) and matrix unit (MXU) that can do numerical computation. Compute units operate on values that live in SREGs and VREGs and output values into those registers as well.
+
+In order to do a vectorized computation on our values `x` and `y` that live in HBM, we need to:
+
+1. Copy the values `x` and `y` into VMEM.
+2. Load the values from VMEM into VREGs.
+3. Execute the computation using the VPU or MXU, storing the output in VREGs.
+4. Store the values in the output VREGs into VMEM.
+5. Copy the output values in VMEM back to HBM.
+
++++ {"id": "TzctMbNsn3vc"}
+
+Let's implement a Pallas function that does just that!
+
+```{code-cell}
+:id: 2IXQxNWrKJyb
+:outputId: d62eb493-5f92-4496-f113-d3cd24cb0b9f
+
+def add_matrices_kernel(x_vmem_ref, y_vmem_ref, z_vmem_ref):
+  # Load x and y from VMEM into VREGs
+  x_vregs = x_vmem_ref[...]
```

Comment:
Maybe use `[:, :]` instead of the rank-polymorphic `[...]`, given the name of the function?